configs/common/packages.yaml

```diff
@@ -72,232 +65,198 @@
-      version: ['1.2.0']
+      require: '@1.2.0'
+    flex:
+      # Pin version to avoid duplicates
+      require: '@2.6.4'
     fms:
-      version: ['2023.04']
-      variants: precision=32,64 +quad_precision +gfs_phys +openmp +pic constants=GFS build_type=Release +deprecated_io
+      variants: 'precision=32,64 +quad_precision +gfs_phys +openmp +pic constants=GFS build_type=Release +deprecated_io'
     fontconfig:
-      variants: +pic
+      require: '+pic'
     freetype:
-      variants: +pic
+      require: '+pic'
     g2:
-      version: ['3.4.9']
+      require: '@3.4.9'
     g2c:
-      version: ['1.6.4']
+      require: '@1.6.4'
     g2tmpl:
-      version: ['1.10.2']
-    gettext:
-      version: ['0.21.1']
+      require: '@1.10.2'
     gfsio:
-      version: ['1.4.1']
-    gftl-shared:
-      version: ['1.6.1']
+      require: '@1.4.1'
     #git-lfs:
       # Assume git-lfs is provided, hard to install
       # because of dependencies on go/go-bootstrap.
       # Note: Uncommenting this entry will break
       # the container builds.
-      #version: ['2.11.0']
+      #require: '@2.11.0'
     grib-util:
-      version: ['1.4.0']
+      require: '@1.4.0'
     gsibec:
-      version: ['1.2.1']
+      require: '@1.2.1'
     gsi-ncdiag:
-      version: ['1.1.2']
+      require: '@1.1.2'
     gsl-lite:
-      version: ['0.37.0']
+      require: '@0.37.0'
     hdf:
-      version: ['4.2.15']
-      variants: +external-xdr ~fortran ~netcdf
+      require: '@4.2.15 +external-xdr ~fortran ~netcdf'
     hdf5:
       require: '@1.14.3 +hl +fortran +mpi +threadsafe ~szip'
     # Newer versions of hdf-eos2 require manual downloading, avoid
     hdf-eos2:
-      require: "@2.20v1.00"
+      require: '@2.20v1.00'
     ip:
-      version: ['5.0.0']
-      variants: precision=4,d,8
+      require: '@5.0.0 precision=4,d,8'
     ip2:
-      version: ['1.1.2']
+      require: '@1.1.2'
     jasper:
-      version: ['2.0.32']
+      require: '@2.0.32'
     jedi-cmake:
-      version: ['1.4.0']
+      require: '@1.4.0'
     landsfcutil:
-      version: ['2.4.1']
+      require: '@2.4.1'
     libjpeg-turbo:
-      version: ['2.1.0']
+      require: '@2.1.0'
     libpng:
-      version: ['1.6.37']
-      variants: +pic
-    libyaml:
-      version: ['0.2.5']
+      require: '@1.6.37 +pic'
     # Newest version of magics needed for oneapi compilers
     magics:
       require: "@4.15.3:"
     mapl:
-      version: ['2.46.2']
-      variants: +shared +pflogger ~f2py
+      require: '@2.46.2 +shared ~f2py'
+      variants: '+pflogger'
     # If making changes here, also check the Discover site configs and the CI workflows
     met:
-      version: ['11.1.0']
-      variants: +python +grib2
+      require: '@11.1.0 +python +grib2'
     metplus:
-      version: ['5.1.0']
+      require: '@5.1.0'
     metis:
-      require: "+int64 +real64"
+      require: '+int64 +real64'
     mpich:
-      variants: ~hwloc +two_level_namespace
+      require: '~hwloc'
```

Comment:
Is that because the two_level_namespace option in mpich no longer exists?

Reply:
Looks like two_level_namespace is always enabled on MacOS so I think we are good.

https://github.com/JCSDA/spack/blob/8ba0bbae6698adf7dce525c0cf72c29edf5fb254/var/spack/repos/builtin/packages/mpich/package.py#L552-L553

Reply:
Perfect, thanks. Not that it matters. We can't use mpich on macOS since it doesn't support mpi_f08.